app/controllers/v0/user_action_events_controller.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+module V0
+  class UserActionEventsController < ApplicationController
+    service_tag 'identity'
+
+    def index
+      user_verification = current_user.user_verification
+
+      start_date = params[:start_date].present? ? Date.parse(params[:start_date]) : 1.month.ago.to_date
+      end_date = params[:end_date].present? ? Date.parse(params[:end_date]) : Time.zone.now
```

Comment:
The `Date.parse(params[:end_date])` will have the same problem as the default time did, if it is set for today's date then it will preclude any results from today from being included since they will be after `00:00:00` hours/minutes/seconds. You should be able to use [`end_of_day`](https://apidock.com/rails/DateTime/end_of_day) to make this be `23:59:59.999999`, I don't think that it will matter that this will end up querying slightly into the future.

```ruby
end_date = params[:end_date].present? ? Date.parse(params[:end_date]).end_of_day : Time.zone.now
```